### PR TITLE
[testing workflow] Disable shards when re-running failed tests

### PR DESCRIPTION
**Pull request by adimoldovan**
### Changes proposed in this Pull Request:

Follow-up of https://github.com/woocommerce/woocommerce/pull/50267
When re-running only the failed Playwright tests using a shard > 1 will run 0 tests.
This is a quick fix by adding the `--shard=1/1` argument together with `--last-failed` to basically disable sharding and ensure tests are actually being run.

### How to test the changes in this Pull Request:

Have a failed test in a shard > 1. Re-run the job. All failed tests and only those should run.

Test runs: 
- [run 1](https://github.com/woocommerce/woocommerce/actions/runs/10305647279/job/28527030109?pr=50492): failed test in shard 6/6
- [run 2](https://github.com/woocommerce/woocommerce/actions/runs/10305647279/job/28527514300?pr=50492#step:9:133): re-run of shard 6/6 - only the failed test ran and it passed
- [run 3](https://github.com/woocommerce/woocommerce/actions/runs/10305647279/job/28527768529?pr=50492): re-run of shard 6/6 after the test passed, all tests ran again

### CODE REVIEW

File: plugins/woocommerce/tests/e2e-pw/playwright.config.js [hunk 1]
```diff
@@ -95,7 +92,11 @@ const config = {
 		{
 			name: 'ui',
 			use: { ...devices[ 'Desktop Chrome' ] },
-			testIgnore: '**/api-tests/**',
+			testMatch: [
+				'**basic.spec.js',
+				'**/product-*spec.js',
+				'**/wordpress-post.spec.js',
+			],
```
Comment (adimoldovan) [inline plugins/woocommerce/tests/e2e-pw/playwright.config.js]:
```suggestion
			testIgnore: '**/api-tests/**',
```

File: plugins/woocommerce/tests/e2e-pw/tests/shopper/wordpress-post.spec.js [hunk 2]
```diff
@@ -8,6 +8,7 @@ test(
 	'logged-in customer can comment on a post',
 	{ tag: [ '@gutenberg', '@payments', '@services' ] },
 	async ( { page } ) => {
+		expect( process.env.GITHUB_RUN_ATTEMPT ).toBe( '2' );
```
Comment (adimoldovan) [inline plugins/woocommerce/tests/e2e-pw/tests/shopper/wordpress-post.spec.js]:
```suggestion
```

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -211,9 +211,11 @@ jobs:
             # Only if there are failed tests, we want to use the --last-failed flag. 
             # The run will fail if we're using the flag and there are no failed tests.
             if [ "$failedTests" -gt 0 ]; then
-              if [ "$LAST_FAILED_RUN" == "1" ]; then
+              if [ "$LAST_FAILED_RUN" == "2" ]; then
```
Comment (adimoldovan) [inline .github/workflows/ci.yml]:
```suggestion
              if [ "$LAST_FAILED_RUN" == "1" ]; then
```

File: plugins/woocommerce/tests/e2e-pw/playwright.config.js [hunk 4]
```diff
@@ -54,9 +53,7 @@ if ( process.env.CI ) {
 	reporter.push( [
 		'html',
 		{
-			outputFolder:
-				PLAYWRIGHT_HTML_REPORT ??
-				`${ testsResultsPath }/reports/playwright-report`,
+			outputFolder: `${ testsResultsPath }/playwright-report`,
```
Comment (adimoldovan) [inline plugins/woocommerce/tests/e2e-pw/playwright.config.js:56]: Unrelated, this fixes an annoying local error.

**Review by tammullen — APPROVED** [review]
Thanks for this and providing example runs to show it action. I can see it works well
`Running 1 test using 1 worker, shard 1 of 1`

**Pull request merged**